### PR TITLE
adding blockID as a property of the link

### PR DESCRIPTION
**Pull request by alexduf**
moving contentId generation logic out of mobile-notifications-content to put it here

### CODE REVIEW

File: src/main/scala/com/gu/mobile.notifications.client/legacy/NotificationBuilder.scala [hunk 1]
```diff
@@ -57,6 +58,22 @@ object NotificationBuilderImpl extends NotificationBuilder {
     )
   )
 
+  private def contentAlertId(cap: ContentAlertPayload): String = {
+    def newContentIdentifier(contentApiId: String): String = s"contentNotifications/newArticle/$contentApiId"
+    def newBlockIdentifier(contentApiId: String, blockId: String): String = s"contentNotifications/newBlock/$contentApiId/$blockId"
+    def contentCoordinates = cap.link match {
```
Comment (agulowaty) [inline src/main/scala/com/gu/mobile.notifications.client/legacy/NotificationBuilder.scala]: could be `val` ?

File: src/main/scala/com/gu/mobile.notifications.client/legacy/NotificationBuilder.scala [hunk 2]
```diff
@@ -57,6 +58,22 @@ object NotificationBuilderImpl extends NotificationBuilder {
...
+      case GuardianLinkDetails(contentApiId, _, _, _, _, blockId) => (Some(contentApiId), blockId)
+      case _ => (None, None)
+    }
+
+    contentCoordinates match {
+      case (Some(contentApiId), Some(blockId)) => newBlockIdentifier(contentApiId, blockId)
```
Comment (agulowaty) [inline src/main/scala/com/gu/mobile.notifications.client/legacy/NotificationBuilder.scala]: could you add spec for this branch ?

File: src/main/scala/com/gu/mobile.notifications.client/legacy/NotificationBuilder.scala [hunk 3]
```diff
@@ -45,7 +46,7 @@ object NotificationBuilderImpl extends NotificationBuilder {
   }
 
   private def buildContentAlert(cap: ContentAlertPayload) = Notification(
-    uniqueIdentifier = cap.id,
+    uniqueIdentifier = contentAlertId(cap),
```
Comment (agulowaty) [inline src/main/scala/com/gu/mobile.notifications.client/legacy/NotificationBuilder.scala]: when you've introduced concept of identifier that spans across two parameters - `contentApiId` and optional `blockId` maybe its time to introduce class for that (named e.g. `ContentIdentifier`) ?

Comment (alexduf) [inline src/main/scala/com/gu/mobile.notifications.client/legacy/NotificationBuilder.scala]: I would if it was straight forward but the Link type is serialized so I don't have much leeway for modification. I prefer to keep it as close as possible from what is going to be serialized.

File: src/main/scala/com/gu/mobile.notifications.client/legacy/NotificationBuilder.scala (hunk 2 above)
Comment (alexduf) [inline src/main/scala/com/gu/mobile.notifications.client/legacy/NotificationBuilder.scala]: Yup :)

**Comment by pvighi** [comment]
Looks alright to me :+1:

**Pull request merged**